src/main/java/org/truffleruby/core/regexp/RegexpNodes.java

```diff
@@ -345,76 +345,6 @@ public static boolean frameIsNotSend(RubyContext context, Object callerFrame) {
-            final DynamicObject dupedString = (DynamicObject) dupNode.call(frame, string, "dup");
-
-            return matchWithStringCopy(frame, regexp, dupedString);
-        }
-
-        @Specialization(guards = "isRubySymbol(symbol)")
-        public Object matchSymbol(VirtualFrame frame, DynamicObject regexp, DynamicObject symbol) {
-            if (toSNode == null) {
-                CompilerDirectives.transferToInterpreterAndInvalidate();
-                toSNode = insert(CallDispatchHeadNode.create());
-            }
-
-            return matchWithStringCopy(frame, regexp, (DynamicObject) toSNode.call(frame, symbol, "to_s"));
-        }
-
-        @Specialization(guards = "isNil(nil)")
-        public Object matchNil(VirtualFrame frame, DynamicObject regexp, Object nil) {
-            return nil();
-        }
-
-        @Specialization(guards = { "!isRubyString(other)", "!isRubySymbol(other)", "!isNil(other)" })
-        public Object matchGeneric(VirtualFrame frame, DynamicObject regexp, DynamicObject other) {
-            if (toStrNode == null) {
-                CompilerDirectives.transferToInterpreterAndInvalidate();
-                toStrNode = insert(ToStrNodeGen.create(null));
-            }
-
-            return matchWithStringCopy(frame, regexp, toStrNode.executeToStr(frame, other));
-        }
-
-        // Creating a MatchData will store a copy of the source string. It's tempting to use a rope here, but a bit
-        // inconvenient because we can't work with ropes directly in Ruby and some MatchData methods are nicely
-        // implemented using the source string data. Likewise, we need to taint objects based on the source string's
-        // taint state. We mustn't allow the source string's contents to change, however, so we must ensure that we have
-        // a private copy of that string. Since the source string would otherwise be a reference to string held outside
-        // the MatchData object, it would be possible for the source string to be modified externally.
-        //
-        // Ex. x = "abc"; x =~ /(.*)/; x.upcase!
-        //
-        // Without a private copy, the MatchData's source could be modified to be upcased when it should remain the
-        // same as when the MatchData was created.
-        private Object matchWithStringCopy(VirtualFrame frame, DynamicObject regexp, DynamicObject string) {
-            final Rope rope = StringOperations.rope(string);
-            final Matcher matcher = createMatcher(getContext(), regexp, rope, bytesNode.execute(rope), true);
-            final int range = rope.byteLength();
-
-            final DynamicObject matchData = matchNode.execute(regexp, string, matcher, 0, range, false);
-            setLastMatchNode.executeSetLastMatch(frame, matchData);
-
-            if (matchData != nil()) {
-                return matcher.getBegin();
-            }
-
-            return nil();
-        }
```

Comment:
This part of the behavior is already handled in `regexp_search_region` it seems.
It would be nice to move the comment there.
And I think that `dup` of the source String should actually be moved to `MatchNode` eventually since that node creates the MatchData.

Reply:
I agree. I think it wasn't in match node originally because it was the wrong side of a truffle boundary. I'll move the comment for now